printing,latex:fixed inconsistency in superscripts raised to power

vedantc98 opened the pull request:
In the case of superscripts like 'x^1' in Latex, when the superscript is raised to a power, the superscript was indistinguished from the power.

As an example, with latex printing enabled:
Initially,
```
M__1, L__1=symbols('M__1, L__1')
a=M__1**3
b=L__1**3
print(a+b)
```
>>'M^{1}^{3}+L^{1}^{3}'

Now, the output is
>>'(M^{1})^{3}+(L^{1})^{3}'

<!-- Please give this pull request a descriptive title. Pull requests with descriptive titles are more likely to receive reviews. Describe what you changed! A title that only references an issue number is not descriptive. -->

<!-- If this pull request fixes an issue please indicate which issue by typing "Fixes #NNNN" below. -->
Fixes #12886

vedantc98 committed:
printing,latex:fixed inconsistency in superscripts raised to power

In the case of superscripts like $$x^1$$ in Latex, when the superscript is raised to a power, the superscript was indistinguished from the power.
As an example, with latex printing enabled:
Initially,
M__1, L__1=symbols('M__1, L__1')
a=M__1**3
b=L__1**3
print(a+b)
>>'M^{1}^{3}+L^{1}^{3}'

Now, the output is
>>'(M^{1})^{3}+(L^{1})^{3}'

asmeurer commented:
Needs to use `\left` and `\right`.

asmeurer commented on sympy/printing/latex.py:
```diff
@@ -460,6 +460,9 @@ def _print_Pow(self, expr):
             base, p, q = self.parenthesize(expr.base, PRECEDENCE['Pow']), expr.exp.p, expr.exp.q
+            #fixes issue #12886, adds parentheses before superscripts raised to powers
+            if '^' in base and expr.base.is_Symbol:
+                base="(%s)"%base
```
Put spaces around the equals sign and % operator.

vedantc98 committed:
printing, latex: code quality and bracket change
Added spaces between operators to maintain code standard in sympy/sympy/printing/latex.py
Changed the brackets to add Latex brackets \left and \right

vedantc98 commented:
Made the necessary additions, changed brackets to \left and \right
Added spaces around operators.

asmeurer commented:
Please add a test for the fix.

vedantc98 committed:
Added test for issue #12886

Added a test for the issue, defined under function test_issue_12886() in sympy/sympy/printing/tests/test_latex.py

vedantc98 commented:
Added the test for fix under sympy/sympy/printing/tests/test_latex.py in function test_issue_12886.py

coderofgames commented:
Hi, I think I found another problem ...

y__1 = Function('y__1')( t ) 
y__2 = Function('y__2')( t ) 
y__3 = Function('y__3')( t )

then 

y__1 **3 + diff(y__2,t) **2 + diff(y__3,t,t) **2 

does not appear the same as if the coordinate is a variable
